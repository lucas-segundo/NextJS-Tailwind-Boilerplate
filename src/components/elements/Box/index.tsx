import { HTMLAttributes } from 'react'

export type BoxProps = {
  children?: React.ReactNode
} & HTMLAttributes<HTMLDivElement>

const Box = ({ children = 'Box', ...props }: BoxProps) => (
  <div {...props} className={`rounded-lg shadow-lg ${props.className}`}>
    {children}
  </div>
)

export default Box
