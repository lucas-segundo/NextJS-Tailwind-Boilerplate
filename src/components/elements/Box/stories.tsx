import { Story, Meta } from '@storybook/react'
import Box, { BoxProps } from '.'

export default {
  title: 'Elements/Box',
  component: Box,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story<BoxProps> = (args) => <Box {...args} />
