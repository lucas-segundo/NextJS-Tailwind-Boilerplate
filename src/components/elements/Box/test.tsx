import { render, screen } from '@testing-library/react'

import Box from '.'

describe('<Box />', () => {
  it('should render correctly', () => {
    const { container } = render(<Box />)

    expect(screen.getByText(/Box/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
