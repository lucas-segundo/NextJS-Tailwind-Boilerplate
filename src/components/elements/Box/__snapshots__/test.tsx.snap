// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render correctly 1`] = `
<div
  class="rounded-lg shadow-lg undefined"
>
  Box
</div>
`;
